src/plots/plots.js

```diff
@@ -1926,7 +1962,34 @@ plots.doCalcdata = function(gd, traces) {
         fullData = gd._fullData,
         fullLayout = gd._fullLayout;
 
-    var trace, _module, i, j;
+    var trace, _module, i, j, index;
+
+    // Look up priorities. The carpet trace is currently
+    var priority;
+    var prioritizedFullDataLookup = [];
+    var hasPriorities = false;
+    for(i = 0; i < fullData.length; i++) {
+        trace = fullData[i];
+        _module = trace._module;
+
+        // Get the priority from the module definition, otherwise default to zero:
+        priority = (_module && _module.calcPriority) ? _module.calcPriority : 0;
+
+        // Note if a non-zero priority was found so that we only need to sort if
+        // there's anything to sort:
+        hasPriorities = hasPriorities || priority;
+
+        prioritizedFullDataLookup.push({index: i, priority: priority});
```

Comment:
Reverting this. I reordered one of the mocks to ensure that calcdata is in an order that would cause it to fail if this were required but not implemented.